[mission2] 2단계 미션 구현 - 이재건/3팀

gunysunny opened the pull request:
## 구현 사항
- 더미데이터가 아닌 TMDB를 이용하여 .env생성하고 API 키값을 받아와 노출함

## 어려웠던 점
- .env를 처음 사용하여서 헤맸다.

## 구현 이미지
- 
![스크린샷 2025-06-19 133352](https://github.com/user-attachments/assets/071aad0f-bdbb-4edb-a545-fcac5b038a2d)
![스크린샷 2025-06-19 133412](https://github.com/user-attachments/assets/76060188-e96f-472d-980c-9043c10495ba)

gunysunny committed:
chore: .env 파일 gitignore에 추가

- 환경변수 파일(.env)이 깃허브에 올라가지 않도록 .gitignore에 추가함

gunysunny committed:
feat: MovieDetail 페이지에서 TMDb API로 영화 상세 정보 조회

- 카드 클릭 시 더미데이터가 아닌 TMDb API를 통해 영화 상세 정보를 받아오도록 변경
- useParams로 id 받아와 fetch로 실시간 데이터 호출

gunysunny committed:
feat: MovieList 페이지에서 TMDb API로 영화 리스트 조회

- 더 이상 더미데이터가 아닌 TMDb API를 사용해 인기 영화 리스트를 받아옴
- fetch 및 상태관리 로직 적용

gunysunny committed:
feat: MovieList 무한스크롤 기능 구현
- 인기 영화 리스트(MovieList)에서 스크롤이 하단에 도달하면 자동으로 추가 영화 목록을 불러오도록 무한스크롤 기능 추가
- Intersection Observer를 사용하여 추가 페이지를 자동으로 불러오도록 구현
- MovieCylinder(캐러셀)는 1페이지 데이터만 사용하고, MovieGrid(그리드)만 무한스크롤 적용
- loading, hasMore 상태 관리로 중복 요청 및 마지막 페이지 처리

Kwonyeojiny commented on .gitignore:
```diff
@@ -22,3 +22,4 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+.env
```
.env 추가 ✅

Kwonyeojiny commented on src/component/moviedetail/MovieDetail.jsx:
```diff
@@ -14,10 +14,22 @@ function MovieDetail() {
-            .then((data) => setMovie(data));     // 파싱된 데이터로 movie 상태 업데이트
-    }, []);
+        const API_TOKEN = import.meta.env.VITE_TMDB_TOKEN; // .env의 v4 토큰
+        const API_URL = `https://api.themoviedb.org/3/movie/${id}?language=ko-KR`;
```
`https://api.themoviedb.org/3/movie` 해당 url도 다른 api 불러 올때마다 사용되는 중복이므로
상수로 빼주면 좋습니당~

gunysunny committed:
refactor: TMDb API 주소 상수로 분리
- 중복되는 TMDb API 기본 URL을 상수(TMDB_MOVIE_BASE_URL)로 선언하여 코드 내 fetch 요청에 일관되게 사용하도록 리팩터링
- 코드 가독성 및 유지보수성 개선

Kwonyeojiny commented on src/component/movielist/MovieList.jsx:
```diff
@@ -18,11 +18,12 @@ function MovieList() {
 
     // loader: 페이지 맨 아래에 붙여서, 화면에 보이면 다음 페이지 불러오도록 감지할 DOM 요소(Ref)
     const loader = useRef(null);
+    const TMDB_MOVIE_BASE_URL = "https://api.themoviedb.org/3/movie";
```
굿입니다!! 👍🏻🔥👍🏻🔥👍🏻🔥

gunysunny committed:
feat: lucide-react 아이콘 패키지 설치 및 package.json 추가

gunysunny committed:
refactor: MovieCylinder 컴포넌트를 3D 원통에서 사선 비틀기 카드 나열 방식으로 변경
